File: java/0010.0_Regular_Expression_Matching.java
/*
执行用时：35 ms, 在所有 Java 提交中击败了10.90% 的用户
内存消耗：41.6 MB, 在所有 Java 提交中击败了5.04% 的用户
通过测试用例：353 / 353
*/

class Solution {
    public boolean isMatch(String s, String p) {
        return s.matches(p);
    }
}


/*
执行用时：32 ms, 在所有 Java 提交中击败了11.51% 的用户
内存消耗：41.5 MB, 在所有 Java 提交中击败了5.04% 的用户
通过测试用例：353 / 353
*/

import java.util.regex.Pattern;

class Solution {
    public boolean isMatch(String s, String p) {
        return Pattern.matches(p, s);
    }
}

File: leetcode-cn/0010.0_Regular_Expression_Matching.py
'''
执行用时：56 ms, 在所有 Python3 提交中击败了37.43% 的用户
内存消耗：14.9 MB, 在所有 Python3 提交中击败了90.15% 的用户
通过测试用例：353 / 353
'''
class Solution:
    def isMatch(self, s: str, p: str) -> bool:
        return bool(re.match('^' + p + '$', s))

'''
执行用时：60 ms, 在所有 Python3 提交中击败了27.61% 的用户
内存消耗：15 MB, 在所有 Python3 提交中击败了81.81% 的用户
通过测试用例：353 / 353
'''
class Solution:
    def isMatch(self, s: str, p: str) -> bool:
        return bool(re.match(p + '$', s))
        